sdk/contracts-provider-service/src/implementation/ContractsProvider.ts

```diff
@@ -0,0 +1,74 @@
+import { IBlockchainClientProvider } from '@summerfi/blockchain-client-provider'
+import { IConfigurationProvider } from '@summerfi/configuration-provider'
+import {
+  IContractsProvider,
+  IErc20Contract,
+  IErc4626Contract,
+  IFleetCommanderContract,
+} from '@summerfi/contracts-provider-common'
+import { IAddress, IChainInfo } from '@summerfi/sdk-common'
+import { Erc20Contract } from './contracts/Erc20Contract/Erc20Contract'
+import { Erc4626Contract } from './contracts/Erc4626Contract/Erc4626Contract'
+import { FleetCommanderContract } from './contracts/FleetCommanderContract/FleetCommanderContract'
+
+/**
+ * @name ContractsProvider
+ * @implements IContractsProvider
+ */
+export class ContractsProvider implements IContractsProvider {
+  private _configProvider: IConfigurationProvider
+  private _blockchainClientProvider: IBlockchainClientProvider
+
+  /** CONSTRUCTOR */
+  constructor(params: {
+    configProvider: IConfigurationProvider
+    blockchainClientProvider: IBlockchainClientProvider
+  }) {
+    this._configProvider = params.configProvider
+    this._blockchainClientProvider = params.blockchainClientProvider
+  }
+
+  /** PUBLIC */
+
+  /** @see IContractsProvider.getErc20Contract */
+  async getErc20Contract(params: {
```

Comment:
why it's an async function?
notice same for all below methods, just didn't wanted to spam the PR